metadata-ingestion/src/datahub/cli/cli_utils.py

```diff
@@ -63,8 +76,50 @@ def get_session_and_host():
 
             gms_host = gms_config.server
             gms_token = gms_config.token
+            return gms_host, gms_token
         except yaml.YAMLError as exc:
             click.secho(f"{DATAHUB_CONFIG_PATH} malformatted, error: {exc}", bold=True)
+    return None, None
+
+
+def get_details_from_env() -> Tuple[Optional[str], Optional[str]]:
```

Comment:
seems like this would be simpler to model as a Dict[str, str] being returned with keys being `GMS_HOST` and `GMS_TOKEN` ..  then you don't need the first-non-null checks later on.. you can just pick the value of these variables from the Dict directly.

Reply:
I would still need to use the `Optional[str]` in `Dict` because it is quite possible one of the variables was set and not the other. We cannot assume user will either provide both of the variables or none. The first non-null check is for that purpose.

Reply:
I'm generally not a fan of using Tuples .. because there is implied order here (first element is host, second element is token) and can lead to bugs later on if someone misunderstands the order ... with a dictionary like "host" -> value, and "token" -> value, it is harder to make those mistakes. 

But I see that the rest of the code in this file has similar things. So I'll let this in and we can do a pass over these later.

Reply:
Good point on the implied order. Had not considered from that point.